tests/foreman/api/test_classparameters.py

```diff
@@ -15,78 +15,197 @@
+    tier1,
+    tier2,
+)
 from robottelo.test import APITestCase
 
 
+@filtered_datapoint
+def valid_sc_parameters_data():
+    """Returns a list of valid smart class parameter types and values"""
+    return [
+        {
+            u'sc_type': 'string',
+            u'value': gen_string('utf8'),
+        },
+        {
+            u'sc_type': 'boolean',
+            u'value': choice(['0', '1']),
+        },
+        {
+            u'sc_type': 'integer',
+            u'value': gen_integer(min_value=1000),
+        },
+        {
+            u'sc_type': 'real',
+            u'value': -123.0,
+        },
+        {
+            u'sc_type': 'array',
+            u'value': "['{0}', '{1}', '{2}']".format(
+                gen_string('alpha'), gen_integer(), gen_boolean()),
+        },
+        {
+            u'sc_type': 'hash',
+            u'value': '{{"{0}": "{1}"}}'.format(
+                gen_string('alpha'), gen_string('alpha')),
+        },
+        {
+            u'sc_type': 'yaml',
+            u'value': 'name=>XYZ',
+        },
+        {
+            u'sc_type': 'json',
+            u'value': '{"name": "XYZ"}',
+        },
+    ]
+
+
+@filtered_datapoint
+def invalid_sc_parameters_data():
+    """Returns a list of invalid smart class parameter types and values"""
+    return [
+        {
+            u'sc_type': 'boolean',
+            u'value': gen_string('alphanumeric'),
+        },
+        {
+            u'sc_type': 'integer',
+            u'value': gen_string('utf8'),
+        },
+        {
+            u'sc_type': 'real',
+            u'value': gen_string('alpha'),
+        },
+        {
+            u'sc_type': 'array',
+            u'value': '0',
+        },
+        {
+            u'sc_type': 'hash',
+            u'value': 'a:test',
+        },
+        {
+            u'sc_type': 'yaml',
+            u'value': '{a:test}',
+        },
+        {
+            u'sc_type': 'json',
+            u'value': gen_string('alpha'),
+        },
+    ]
+
+
 class SmartClassParametersTestCase(APITestCase):
     """Implements Smart Class Parameter tests in API"""
 
+    @classmethod
+    def setUpClass(cls):
+        super(SmartClassParametersTestCase, cls).setUpClass()
+        cls.host_name = settings.server.hostname
+        cls.host = entities.Host().search(
+            query={'search': 'name=' + cls.host_name}
+        )[0]
+        ssh.command('puppet module install --force puppetlabs/ntp')
+        cls.env = entities.Environment().search(
+            query={'search': 'name="production"'}
+        )[0]
+        cls.proxy = entities.SmartProxy(name=cls.host_name).search()[0]
+        cls.proxy.import_puppetclasses(environment=cls.env)
+        cls.puppet = entities.PuppetClass().search(
+            query={'search': 'name="ntp"'}
+        )[0]
+        cls.sc_params_list = entities.SmartClassParameters().search(
+            query={'search': 'puppetclass="ntp"', 'per_page': 100}
+        )
+        if len(cls.sc_params_list) < 45:
```

Comment:
I think we need to figure out a better, clever way to do this as I'm not fond of writing tests that depend on hard coded, expected "dependencies". Is there any way to extract the information from the module so that we can make this more dynamic?